app/assets/javascripts/pretty-text/engines/discourse-markdown/d-wrap.js.es6

```diff
@@ -0,0 +1,47 @@
+import { parseBBCodeTag } from "pretty-text/engines/discourse-markdown/bbcode-block";
+
+function addWrapper(buffer, matches, state) {
+  let token;
+
+  let parsed = parseBBCodeTag(
+    "[wrap wrap" + matches[1] + "]",
+    0,
+    matches[1].length + 11
+  );
+
+  token = new state.Token("div_open", "div", 1);
+  token.attrs = [["class", "d-wrap"]];
+
+  const attributes = parsed.attrs || {};
+  const content = attributes.content;
+  delete attributes.content;
+
+  Object.keys(attributes).forEach(tag => {
+    const value = attributes[tag];
+    token.attrs.push([`data-${tag}`, state.md.utils.escapeHtml(value)]);
```

Comment:
I assume this can't be used for injection right? The parser is smart enough to remove quotes and such?

I would feel better if the attribute name was limited to a regexp, say `[a-z0-9\-]` ?

Reply:
I think the spec is rather safe on this and browsers would as you assume get rid of any malicious attempt.

For extra safety I'm also escaping attribute and added two specs to prevent basic regression on this area.